pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -333,23 +333,20 @@ func GetHugepageLimitsFromResources(resources v1.ResourceRequirements) []*runtim
 	}
 
 	requiredHugepageLimits := map[string]uint64{}
-	for resourceObj, amountObj := range resources.Limits {
-		if !v1helper.IsHugePageResourceName(resourceObj) {
-			continue
-		}
 
-		pageSize, err := v1helper.HugePageSizeFromResourceName(resourceObj)
-		if err != nil {
-			klog.InfoS("Failed to get hugepage size from resource", "object", resourceObj, "err", err)
-			continue
+	// When hugepage limits are specified at pod level, the container limit will
+	// match the pod level limit, however if the container also specifies,
+	// its limit will have precedence over the pod level limit
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodLevelResources) && resourcehelper.IsPodLevelResourcesSet(pod) {
+		for limitName, limitAmount := range pod.Spec.Resources.Limits {
+			readAndDefineRequiredHugepageLimit(requiredHugepageLimits, limitName, limitAmount)
 		}
+	}
 
-		sizeString, err := v1helper.HugePageUnitSizeFromByteSize(pageSize.Value())
-		if err != nil {
-			klog.InfoS("Size is invalid", "object", resourceObj, "err", err)
-			continue
-		}
-		requiredHugepageLimits[sizeString] = uint64(amountObj.Value())
+	// If the container and the pod specify, the container will have precedence,
```

Comment:
this comment seems confusing. It sounds like pod-level limit is being overriden.

Reply:
What I meant is that the container limits will have precedence over the pod level limits in regards to the cgroup values (overriding the cgroups obtained from the pod level resources in the previous for loop), if container level hugepage limits are specified.

So I removed the overriding terminology and updated the pod level comment to:
```
	// When hugepage limits are specified at pod level and no hugepage limits are
	// specified at container level, the container's cgroup will reflect the pod level limit.
```

And the container level comment to:
```
	// If both the pod level and the container level specify hugepages, the container
	// level will have precedence, so the container's hugepages limit will be reflected
	// in the container's cgroup values.
```